Sources/GRPCCodeGen/SourceGenerator.swift

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// Creates a ``SourceFile`` containing the generated code for the RPCs represented in a ``CodeGenerationRequest`` object.
+public struct SourceGenerator {
+  /// The options regarding the access level, indentation for the generated code
+  /// and whether to generate server and client code.
+  public var configuration: Configuration
+
+  public init(configuration: Configuration) {
+    self.configuration = configuration
+  }
+
+  /// User options for the CodeGeneration.
+  public struct Configuration: Sendable {
+    /// The access level the generated code will have.
+    ///
+    /// - SeeAlso: ``AccessLevel-swift.struct``.
```

Comment:
No point adding the see also here, the generated docs will already link to `AccessLevel`